src/descriptor/mod.rs

```diff
@@ -196,6 +256,17 @@ impl fmt::Display for DescriptorPublicKey {
     }
 }
 
+impl DescriptorSecretKey {
+    pub fn as_public(&self) -> Result<DescriptorPublicKey, DescriptorKeyParseError> {
```

Comment:
Needs a doccomment, and should mention that it will drop some of the derivation path if this involves hardened derivations. (Maybe it should be renamed to reflect this, but I'm not sure what a better name would be.)

Reply:
Can you rebase so that the doccomment was there originally? Otherwise the individual commits do not compile.

You can check whether the commits compile by running `git rebase -x 'cargo test && cargo +1.29.0 test' master` on your branch.

Reply:
It should be good now..